src/pokeradar.c

```diff
@@ -0,0 +1,590 @@
+    u8 v3;
+    u8 v4;
+    int v5, v6;
+    u8 v7;
+    const VecFx32 * v8;
+    u8 ringTileCount[4] = {  // Number of tiles in each ring of the radar. Lowest being the most outer ring
+        32, 24, 16, 8
+    };
+
+    v8 = sub_0205EAFC(param0->unk_3C);
+
+    v7 = 0;
+
+    for (patchRing = 0; patchRing < NUM_GRASS_PATCHES; patchRing++) {
+        BOOL v10;
+
+        v3 = inline_020564D0(ringTileCount[patchRing]);
+
+        v1 = 9 - (patchRing * 2);
+
+        v2 = 9 - (patchRing * 2);
+
+        v4 = v3 / v1;
+
+        if (v4 == 0) {
+            v5 = patchRing + v3 % v1;
+            v6 = patchRing;
+        } else if (v4 == 1) {
+            v5 = patchRing + v3 % v1;
+            v6 = patchRing + v2 - 1;
+        } else {
+            GF_ASSERT(v3 >= (v1 * 2));
+            v4 = v3 - (v1 * 2);
+
+            v6 = patchRing + (v4 / 2) + 1;
+
+            if (v4 % 2 == 0) {
+                v5 = patchRing;
+            } else {
+                v5 = patchRing + v1 - 1;
+            }
+        }
+
+        v10 = CheckTileIsGrass(param0, v8->y, param1, param2, v5, v6, &chain->patch[patchRing]);
+
+        if (v10) {
+            v7++;
+        }
+    }
+
+    if (v7 == 0) {
+        ClearRadarChain(chain);
+
+        sub_02055554(param0, sub_02055428(param0, param0->unk_1C->unk_00), 1);
+    } else {
+        chain->active = TRUE;
+    }
+
+    return chain->active;
+}
+
+void SetupGrassPatches (UnkStruct_0203CDB0 * param0, const int param1, RadarChain * chain) {
+    u8 patchRing;
+
+    for (patchRing = 0; patchRing < NUM_GRASS_PATCHES; patchRing++) {
+        if (chain->patch[patchRing].active) {
+            chain->patch[patchRing].continueChain = CheckPatchContinueChain(patchRing, param1);
+
+            if (!chain->patch[patchRing].continueChain) {
+                if (inline_020564D0(100) < 50) {  // If the patch will break the chain, it has a 50/50 chance of shaking the other type
+                    chain->patch[patchRing].shakeType = 0;
+                } else {
+                    chain->patch[patchRing].shakeType = 1;
+                }
+
+                chain->patch[patchRing].shiny = FALSE;
+            } else {
+                chain->patch[patchRing].shakeType = chain->shakeType;  // A patch that continues the chain, shakes the type the chain is set to
+
+                chain->patch[patchRing].shiny = CheckPatchShiny(param0->chain->count);
+            }
+        }
+    }
+}
+
+void sub_02069638 (UnkStruct_0203CDB0 * param0, RadarChain * chain)
+{
+    u8 patchRing;
+
+    for (patchRing = 0; patchRing < NUM_GRASS_PATCHES; patchRing++) {
+        if (chain->patch[patchRing].active) {
+            int v1 = chain->patch[patchRing].unk_00;
+            int v2 = chain->patch[patchRing].unk_04;
+
+            if (chain->patch[patchRing].shiny) {
+                chain->patch[patchRing].unk_18 = ov5_021F3154(param0, v1, v2, 2);
+            } else {
+                if (chain->patch[patchRing].shakeType == 0) {
```

Comment:
Should this `0` belong to an `enum`, eventually?

Reply:
I think it has 2 values, it probably wouldn't hurt to make it part of an enum. I'd have to check which one is which in game

Reply:
verified in game